### ISSUE TITLE
`pswp--zoomed-in` should be added immediately if zooming

### ISSUE THREAD

**Issue by hirasso** [dimsemenov/PhotoSwipe#1987]
Hi there, first I must say again how much I love this library!! Congrats on the new major version 🎉

Today I migrated one of my projects from v4 to v5 and I noticed that the class `pswp--zoomed-in` is not being added before the zoom-animation finishes. This used to be different in v4, where the class was being added immediately after clicking an image. I rely on this class to hide a custom element that I render as an overlay for my images. 

As a side note: While looking through the related code, I noticed [this line](https://github.com/dimsemenov/PhotoSwipe/blob/master/src/js/ui/ui.js#L21):
```js
el.classList[isZoomedIn ? 'add' : 'remove']('pswp--zoomed-in');
```
Is there a reason you are not using [`classList.toggle`](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#examples) here? Seems like the perfect use case :) The above could be refactored like this:
```js
el.classList.toggle('pswp--zoomed-in', isZoomedIn);
```

**Comment by hirasso** [comment]
As a temporary workaround, I used this code to add the class right before zooming:
```js
const lightbox = new PhotoSwipeLightbox(options);
lightbox.on("beforeZoomTo", (photoSwipeEvent) => {
  // bail early if the class is already there
  if (lightbox.pswp.element.classList.contains("pswp--zoomed-in")) return;

  const { currZoomLevel } = lightbox.pswp.currSlide;
  const { destZoomLevel } = photoSwipeEvent;

  // bail early if zooming out
  if (currZoomLevel > destZoomLevel) return;
  
  lightbox.pswp.element.classList.add("pswp--zoomed-in");
});
```
I still have the feeling there would be room for improvement here.

**Comment by dimsemenov** [comment]
The v4 was changing class directly in `zoomTo` method, while the new one is based on the actual zoom value in API.  Since `zoomTo` uses CSS animations, you can't reliably get zoom value mid-transition, so it updates when the animation ends.

I like your solution though, should probably be merged, unless we find some issue with it.

> Is there a reason you are not using classList.toggle here? Seems like the perfect use case :)

Think it just was ported from some older code that had different browser support, should be changed as well.

**Comment by hirasso** [comment]
Great to hear, looking forward!

**Comment by dimsemenov** [comment]
Of course, feel free to submit any changes to `src/`, I'll review and accept them if they make sense.

**Comment by hirasso** [comment]
Sorry, I over-estimated my available time and will not be able to contribute to the code base in the near future. Just thought I would give you a short notice, instead of just ghosting 👻
